server.go

```diff
@@ -383,35 +366,38 @@ func (s *server) Stop() error {
 	s.discoverSrv.Stop()
 	s.cc.wallet.Shutdown()
 	s.cc.chainView.Stop()
+	s.connMgr.Stop()
 
-	// Signal all the lingering goroutines to quit.
-	close(s.quit)
+	peers := s.Peers()
+	for _, peer := range peers {
+		s.DisconnectPeer(peer.addr.IdentityKey)
+	}
+
+	// Wait for all lingering goroutines to quit.
 	s.wg.Wait()
 
 	return nil
 }
 
+func (s *server) Stopped() bool {
+	return atomic.LoadInt32(&s.shutdown) != 0
+}
+
 // genNodeAnnouncement generates and returns the current fully signed node
 // announcement. If refresh is true, then the time stamp of the announcement
 // will be updated in order to ensure it propagates through the network.
 func (s *server) genNodeAnnouncement(refresh bool) (*lnwire.NodeAnnouncement, error) {
-	s.annMtx.Lock()
-	defer s.annMtx.Unlock()
+	s.mu.Lock()
+	defer s.mu.Unlock()
 
 	if !refresh {
 		return s.currentNodeAnn, nil
 	}
 
 	var err error
-
-	newStamp := uint32(time.Now().Unix())
-	if newStamp <= s.currentNodeAnn.Timestamp {
-		newStamp = s.currentNodeAnn.Timestamp + 1
-	}
-
-	s.currentNodeAnn.Timestamp = newStamp
-	s.currentNodeAnn.Signature, err = discovery.SignAnnouncement(
-		s.nodeSigner, s.identityPriv.PubKey(), s.currentNodeAnn,
+	s.currentNodeAnn.Timestamp = uint32(time.Now().Unix())
+	s.currentNodeAnn.Signature, err = discovery.SignAnnouncement(s.nodeSigner,
+		s.identityPriv.PubKey(), s.currentNodeAnn,
```

Comment:
The prior diff here is correct. Looks like a slight mistake introduced during a rebase.